commons/com.b2international.index/src/com/b2international/index/query/Expressions.java

```diff
@@ -315,6 +298,56 @@ public static QueryStringExpression queryString(String query) {
 	public static QueryStringExpression queryString(String query, String defaultField) {
 		return new QueryStringExpression(query, defaultField);
 	}
-	
+
+	public static Expression matchDynamic(String field, Collection<String> dynamicFieldFilters) {
+		ExpressionBuilder bool = bool();
+		for (String dynamicFieldFilter : dynamicFieldFilters) {
+			if (dynamicFieldFilter.contains(DYNAMIC_VALUE_DELIMITER)) {
+				final String propertyName = dynamicFieldFilter.split(DYNAMIC_VALUE_DELIMITER)[0];
+				final String propertyValue = dynamicFieldFilter.substring(propertyName.length() + 1, dynamicFieldFilter.length());
```

Comment:
Minor nitpick: since you already need the position of the separator, an `indexOf + substring + substring` might be slightly faster than a `contains + split[0] + substring` series of operations.